openeo/rest/connection.py

```diff
@@ -833,7 +833,7 @@ def load_collection(
         """
         Load a DataCube by collection id.
 
-        :param collection_id: image collection identifier
```

Comment:
Should this be deprecated instead of being removed?

Reply:
Would I just put `@deprecated("Use 'id' instead") above the `:param collection_id` line?

Reply:
Sounds good to me, but @soxofaan should have the final vote.

Reply:
I don't think we can make a breaking change like this.
Existing code that uses keyword argument `load_collection(collection_id="S2")` will break.

There should be a fallback mechanism:
- give `id` default value None
- add a `**kwargs` and check if `collection_id` is set (and raise deprecation warning if that is the case)

Reply:
Yes, that's what I meant to say :-D

Reply:
@soxofaan
```python
def load_collection(
    self,
    id: str = None,
[...]
    if "collection_id" in kwargs:
        id = kwargs["collection_id"]
        warnings.warn("The use of `collection_id` is deprecated, use `id` instead.", DeprecationWarning)
```
like so?